src/deploy/campaign/deploy-campaign.ts

```diff
@@ -0,0 +1,85 @@
+import { ICampaignArgs, ICampaignState, IDeployCampaignConfig } from "./types";
+import { Deployer } from "../deployer/deployer";
+import { TDeployMissionCtor } from "../missions/types";
+import { BaseDeployMission } from "../missions/base-deploy-mission";
+import { Contract } from "ethers";
+import { BaseStorageAdapter } from "../storage/base-storage-adapter";
+
+
+export class DeployCampaign {
+  state : ICampaignState;
+  deployer : Deployer;
+  // TODO dep: fix typing
+  dbAdapter : BaseStorageAdapter;
+  logger : Console;
+  config : IDeployCampaignConfig;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [name : string | symbol] : any;
+
+  private static indexedHandler : ProxyHandler<DeployCampaign> = {
+    get: (target, prop) => {
+      if (typeof prop === "string" && !!target.state.contracts[prop]) {
```

Comment:
I believe in this case there's no need to use the double not `!!`, as it would be checked for its truth value anyways in the `if`. Same for some of the other uses in this file

Reply:
this is more thorough. converting the value into a boolean. this is never bad usually, while just relying for it to check the value presence might not always work, in theory

Reply:
If the value isn't present it will just be `undefined` and so will resolve to `false`

While it doesn't do any harm to have the additional `!!` it doesn't make a difference if it's in an. `if(variable)` and `if(!!variable)` would be equivalent whereas `const x = variable` wouldn't be equivalent with `const x = !!variable`